### PR TITLE
undmg: 1.0.2 -> 1.0.3

### PR DESCRIPTION
**Pull request by ndowens**
###### Motivation for this change
Update

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/archivers/undmg/default.nix [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
-    sha256 = "0w9vwvj9zbpsjkg251bwv9y10wjyjmh54q2piklz74w64rlbqblr";
-    name = "undmg-${version}";
+    rev = "v${version}";
+    sha256 = "1pxqw92h2w75d4jwiihwnkhnsfk09cddh3flgrqwh9r3ry14fgbb";
+    
   };
 
   buildInputs = [ zlib bzip2 ];
 
-  setupHook = ./setup-hook.sh;
```
Comment (LnL7) [inline pkgs/tools/archivers/undmg/default.nix]: Why did you remove the `setupHook`? It enables packages fetch `dmg` sources when this package is added to their buildInputs (eg. [musescore](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/audio/musescore/darwin.nix#L14-L19)).

File: pkgs/tools/archivers/undmg/default.nix [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
+  makeFlags = "PREFIX=$(out)";
 
-  meta = {
+  meta = with stdenv.lib; {
     homepage = https://github.com/matthewbauer/undmg;
     description = "Extract a DMG file";
-    license = stdenv.lib.licenses.gpl3;
-    platforms = stdenv.lib.platforms.all;
+    license = licenses.gpl3;
+    platforms = platforms.all;
```
Comment (LnL7) [inline pkgs/tools/archivers/undmg/default.nix]: You can add me to the maintainers, and perhaps also @matthewbauer since it's his package.

**Comment by ndowens** [comment]
Didn't think it was needed but will add back

File: pkgs/tools/archivers/undmg/default.nix (hunk 2 above)
Comment (ndowens) [inline pkgs/tools/archivers/undmg/default.nix]: Done

**Review by LnL7 — APPROVED** [review]
